### PR TITLE
fix dependencies that used to be build

### PR DESCRIPTION
**Pull request by deads2k**
This removes dead deps, bumps the API, and makes a minimal change that can get us to merge.

/assign @adambkaplan

### CODE REVIEW

File: pkg/build/controller/build/defaults/defaults.go [hunk 1]
```diff
@@ -79,11 +79,14 @@ func setPodLogLevelFromBuild(pod *corev1.Pod, build *buildv1.Build) error {
 }
 
 func (b BuildDefaults) applyPodDefaults(pod *corev1.Pod, isCustomBuild bool) {
-	if len(b.Config.NodeSelector) != 0 && pod.Spec.NodeSelector == nil {
-		// only apply nodeselector defaults if the pod has no nodeselector labels
-		// already.
+	// TODO this is needs to be fixed properly
```
Comment (deads2k) [inline pkg/build/controller/build/defaults/defaults.go]: @adambkaplan this indicates that something is likely to be broken in the config mapping, but we need something to land the rest of the bumps.

File: pkg/build/controller/build/defaults/defaults.go [hunk 2]
```diff
@@ -79,11 +79,14 @@ func setPodLogLevelFromBuild(pod *corev1.Pod, build *buildv1.Build) error {
...
+	if b.Config.NodeSelector != nil && len(b.Config.NodeSelector.MatchExpressions) > 0 {
+		panic("adambkaplan needs to fix this branch")
```
Comment (soltysh) [inline pkg/build/controller/build/defaults/defaults.go]: lol

Comment (adambkaplan) [reply pkg/build/controller/build/defaults/defaults.go]: is `pod.Affinity` our best option to handle match expressions?

**Pull request merged**